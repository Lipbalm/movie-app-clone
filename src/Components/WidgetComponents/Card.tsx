import React, { FC, useState } from "react";
import { ITopRatedResults } from "../../Modules/Interfaces";
import {
  RectCard,
  ThumnailWrapper,
  ThumnailImg
} from "../../Styles/WidgetStyle";
import { baseImageURL } from "../../Modules/apis";
import CardLabel from "./CardLabel";

interface IProps {
  card_info: ITopRatedResults;
  index: number;
}

const Card: FC<IProps> = ({ card_info, index }) => {
  const { id, genre_ids, title, poster_path } = card_info;

  const [value, setValue] = useState<string>("");

  const handleClick = () => {
    setValue("Clicked");
  };

  return (
    <RectCard
      isMargin={index > 0 ? true : false}
      data-testid="card"
      onClick={handleClick}
    >
      <ThumnailWrapper>
        <ThumnailImg src={`${baseImageURL}${poster_path}`} alt="no image" />
      </ThumnailWrapper>
      <CardLabel genre_ids={genre_ids} title={title} />
    </RectCard>
  );
};

export default Card;
