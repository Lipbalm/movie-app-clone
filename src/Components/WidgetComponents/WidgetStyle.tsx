import styled from "styled-components";
import { Link } from "react-router-dom";
import { IStyleFrame, IStyle } from "../../Modules/StyleInterfaces";
import { combineStyle } from "../../Modules/utils";

const widgetWrapperStyle: IStyleFrame = {
  basis: {
    "margin": "auto",
    "padding-top": "15px",
    "width": "100%"
  }
};

export const WidgetWrapper = styled.div<IStyle>`
  ${props => combineStyle(widgetWrapperStyle, props.styleObj)}
`;

const widgetHead: IStyleFrame = {
  basis: {
    "display": "flex",
    "align-items": "center"
  }
};

export const WidgetHead = styled.h2<IStyle>`
  ${props => combineStyle(widgetHead, props.styleObj)}
`;

const widgetLinkStyle: IStyleFrame = {
  basis: {
    "color": "#797979",
    "font-size": "1.1rem",
    "margin-left": "auto",
    "display": "-webkit-flex",
    "-webkit-align-items": "center",
    "align-items": "center",
    "font-weight": "300"
  }
};

export const WidgetLink = styled(Link)<{ styleObj?: IStyleFrame }>`
  ${props => combineStyle(widgetLinkStyle, props.styleObj)}
`;

const widgetContentsStyle: IStyleFrame = {
  basis: {
    "display": "flex",
    "width": "100%",
    "padding-top": "24px"
  }
};

export const WidgetContents = styled.div<IStyle>`
  ${props => combineStyle(widgetContentsStyle, props.styleObj)}
`;

const widgetContentsInnerStyle: IStyleFrame = {
  basis: {
    "display": "flex",
    "flex-wrap": "wrap",
    "width": "100%",
    "justify-content": "space-between"
  }
};

export const WidgetContentsInner = styled.div<IStyle>`
  ${props => combineStyle(widgetContentsInnerStyle, props.styleObj)}
`;

const WidgetContentsColumnStyle: IStyleFrame = {
  basis: {
    "display": "block",
    "flex-wrap": "wrap",
    "margin": "0 0 0 0"
  }
};

export const WidgetContentsColumn = styled.div<IStyle>`
  ${props => combineStyle(WidgetContentsColumnStyle, props.styleObj)}
`;

const rectCardStyle: IStyleFrame = {
  basis: {
    "width": "13.75rem",
    "box-shadow": "0 1px 11px 0 rgba(0, 0, 0, 0.1)",
    "transition": "box-shadow 0.3s ease-in"
  },
  pesudo: {
    ":hover": {
      "cursor": "pointer",
      "box-shadow": "0 1px 11px 0 rgba(0, 0, 0, 0.5)"
    }
  }
};

export const RectCard = styled.div<IStyle>`
  ${props => combineStyle(rectCardStyle, props.styleObj)}
`;

const themeItemWrapperStyle: IStyleFrame = {
  basis: {
    "display": "flex",
    "margin": "0 0 0 0",
    "box-shadow": "0 1px 11px 0 rgba(0, 0, 0, 0.1)",
    "transition": "box-shadow 0.3s ease-in"
  },
  pesudo: {
    ":hover": {
      "cursor": "pointer",
      "box-shadow": "0 1px 11px 0 rgba(0, 0, 0, 0.5)"
    }
  }
};
export const ThemeItemWrapper = styled.div<IStyle>`
  ${props => combineStyle(themeItemWrapperStyle, props.styleObj)}
`;

const thumnailWrapper: IStyleFrame = {
  basis: {
    "position": "relative",
    "overflow": "hidden"
  }
};
export const ThumnailWrapper = styled.div<IStyle>`
  ${props => combineStyle(thumnailWrapper, props.styleObj)}
`;

const thumnailStyle: IStyleFrame = {
  basis: {
    "background-size": "contain",
    "position": "absolute",
    "background-repeat": "no-repeat",
    "box-sizing": "inherit",
    "width": "inherit",
    "height": "inherit"
  }
};
export const Thumnail = styled.div<IStyle>`
  ${props => combineStyle(thumnailStyle, props.styleObj)}
`;

const themeItemLayerStyle: IStyleFrame = {
  basis: {
    "position": "absolute"
  }
};
export const ThemeItemLayer = styled.div<IStyle>`
  ${props => combineStyle(themeItemLayerStyle, props.styleObj)}
`;

const bottomTagWrapperStyle: IStyleFrame = {
  basis: {
    "display": "block",
    "padding": "1rem 1rem 1.75rem",
    "background-color": "transparent"
  }
};

export const BottomTagWrapper = styled.div<IStyle>`
  ${props => combineStyle(bottomTagWrapperStyle, props.styleObj)}
`;

const overlayTagWrapperStyle: IStyleFrame = {
  basis: {
    "position": "absolute",
    "width": "80%",
    "bottom": "0",
    "padding": "0",
    "background-color": "transparent"
  }
};
export const OverlayTagWrapper = styled.div<IStyle>`
  ${props => combineStyle(overlayTagWrapperStyle, props.styleObj)}
`;

const bottomGenreTagStyle: IStyleFrame = {
  basis: {
    "display": "block",
    "font-size": "0.9rem",
    "height": "1.5rem",
    "margin-bottom": "4px"
  }
};

export const BottomGerenTag = styled.span<IStyle>`
  ${props => combineStyle(bottomGenreTagStyle, props.styleObj)}
`;

const bottomTitleTagStyle: IStyleFrame = {
  basis: {
    "display": "block",
    "font-size": "1.15rem",
    "font-weight": "600",
    "text-overflow": "ellipsis",
    "overflow": "hidden",
    "white-space": "nowrap"
  }
};

export const BottomTitleTag = styled.a<IStyle>`
  ${props => combineStyle(bottomTitleTagStyle, props.styleObj)}
`;

const overlayTitleTagStyle: IStyleFrame = {
  basis: {
    "display": "block",
    "color": "white",
    "font-weight": "600",
    "text-overflow": "ellipsis",
    "overflow": "hidden",
    "white-space": "nowrap"
  }
};

export const OverlayTitleTag = styled.a<IStyle>`
  ${props => combineStyle(overlayTitleTagStyle, props.styleObj)}
`;

const separaterStyle: IStyleFrame = {
  basis: {
    "width": "100%",
    "height": "2px",
    "background-color": "#ebdddd",
    "margin-top": "1rem",
    "position": "relative"
  }
};
export const Separater = styled.div<IStyle>`
  ${props => combineStyle(separaterStyle, props.styleObj)}
`;

const segmentStyle: IStyleFrame = {
  basis: {
    "width": "7rem",
    "height": "4px",
    "background-color": "#e34d4d",
    "top": "-1px",
    "position": "absolute"
  }
};
export const Segment = styled.div<IStyle>`
  ${props => combineStyle(segmentStyle, props.styleObj)}
`;
