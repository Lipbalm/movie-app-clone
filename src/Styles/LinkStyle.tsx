import styled from "styled-components";
import { Link } from "react-router-dom";

const HeaderLinkWrapper = styled.div`
  padding-left: 1rem;
  padding-right: 1rem;
  position: relative;
  width: 1200px;
  margin: 0px auto;
  display: flex;
  -moz-box-align: center;
  align-items: center;
  height: 5rem;
`;

const HeaderList = styled.div`
  display: flex;
`;

const HeaderItem = styled.div`
  margin-left: 20px;
`;

const HeaderLink = styled(Link)`
  font-weight: 800;
  color: black;
  :hover {
    color: green;
  }
`;

export { HeaderLinkWrapper, HeaderList, HeaderItem, HeaderLink };
