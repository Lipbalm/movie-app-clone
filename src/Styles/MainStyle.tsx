import styled, { createGlobalStyle } from "styled-components";
import { Link } from "react-router-dom";

const AppContainer = styled.div`
  width: inherit;
  height: inherit;
`;

const HeaderContainer = styled.div`
  position: fixed;
  left: 0px;
  top: 0px;
  right: 0px;
  z-index: 10;
`;

const RouterHomeContainer = styled.div`
  /* border-style: groove; */
  margin-top: 80px;
`;

const CarouselContainer = styled.div`
  width: inherit;
`;

const BodyWrapper = styled.div`
  margin: 0px auto;
  width: 1200px;
`;

const GlobalStyle = createGlobalStyle`
    body{
        padding: 0;
        margin: 0;
        /* width:100%;
        height:100%; */
    }
    a{
      text-decoration: none;
    }
`;

export {
  AppContainer,
  HeaderContainer,
  RouterHomeContainer,
  CarouselContainer,
  GlobalStyle,
  BodyWrapper
};
