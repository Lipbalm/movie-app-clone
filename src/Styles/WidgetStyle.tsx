import styled from "styled-components";
import { Link } from "react-router-dom";
import { VerticalItemProps } from "../Modules/Interfaces";

const CardWidgetWrapper = styled.div`
  margin: auto;
  padding-top: 15px;
`;

const WidgetHead = styled.h2`
  display: flex;
  align-items: center;
`;

const WidgetLink = styled(Link)`
  color: #797979;
  font-size: 1.1rem;
  margin-left: auto;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  font-weight: 300;
`;

const WidgetContents = styled.div``;

const WidgetContentsInner = styled.div`
  display: flex;
  flex-wrap: wrap;
  font-size: 40px;
  width: 100%;
`;

const RectCard = styled.div<VerticalItemProps>`
  width: 5.5em;
  height: 9.5em;
  box-shadow: 0 1px 11px 0 rgba(0, 0, 0, 0.1);
  transition: box-shadow 0.3s ease-in;
  margin-left: ${props => (props.isMargin ? "20px" : "0px")};
`;

const WidgetWarn = styled.div`
  text-align: center;
  width: inherit;
  height: inherit;
  font-size: 8px;
`;

const ThumnailWrapper = styled.div`
  width: inherit;
  height: 282px;
  position: relative;
  overflow: hidden;
`;

const ThumnailImg = styled.img`
  text-align: center;
  width: inherit;
  height: 282px;
`;

const CardLabelContainer = styled.div`
  display: block;
  padding: 1rem 1rem 1.75rem;
`;

const GerenTag = styled.span`
  display: block;
  font-size: 0.9rem;
  height: 1.5rem;
  margin-bottom: 4px;
`;

const TitleTag = styled.a`
  display: block;
  font-size: 1.15rem;
  font-weight: 600;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`;

export {
  CardWidgetWrapper,
  WidgetHead,
  WidgetLink,
  WidgetContents,
  WidgetContentsInner,
  RectCard,
  WidgetWarn,
  ThumnailWrapper,
  ThumnailImg,
  CardLabelContainer,
  GerenTag,
  TitleTag
};
