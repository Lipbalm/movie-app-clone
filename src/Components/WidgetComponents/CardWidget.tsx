import React, { FC } from "react";
import {
  CardWidgetWrapper,
  WidgetHead,
  WidgetLink,
  WidgetContents,
  WidgetContentsInner
} from "../../Styles/WidgetStyle";
import { ITopRatedResults } from "../../Modules/Interfaces";
import Card from "./Card";

interface IProp {
  head_text: string;
  cards: ITopRatedResults[];
}

const CardWidget: FC<IProp> = ({ head_text, cards }) => {
  return (
    <CardWidgetWrapper>
      <WidgetHead>
        {head_text}
        <WidgetLink to="/theme">더보기</WidgetLink>
      </WidgetHead>
      <WidgetContents>
        <WidgetContentsInner>
          {cards.map(
            (v, i) => (
              <Card card_info={v} index={i} key={i} />
            ),
            0
          )}
        </WidgetContentsInner>
      </WidgetContents>
    </CardWidgetWrapper>
  );
};

export default CardWidget;
