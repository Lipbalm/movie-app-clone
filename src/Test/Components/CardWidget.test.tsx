import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { createMemoryHistory } from "history";
import CardWidget from "../../Components/WidgetComponents/CardWidget";
import { Router } from "react-router-dom";
import { tempCards } from "../TempData/Card";
import { CardWidgetWrapper } from "../../Styles/WidgetStyle";

const setup = () => {
  const history = createMemoryHistory();
  const utils = render(
    <Router history={history}>
      <CardWidget head_text="2020년 1분기 기대신작 애니" cards={tempCards} />
    </Router>
  );
  return { ...utils };
};

describe("<CardWidget />", () => {
  it("snapshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("CardWidget HeadText", () => {
    const { getByText } = setup();
    getByText("2020년 1분기 기대신작 애니");
  });

  it("CardWidget HeadLink", () => {
    const { getByText } = setup();
    getByText("더보기");
  });

  it("CardWidget get Contents", () => {
    const { getAllByTestId } = setup();
    const cards = getAllByTestId("card");

    cards.forEach((v, i) => {
      expect(v).toHaveTextContent(tempCards[i].title);
    });
  });
});
