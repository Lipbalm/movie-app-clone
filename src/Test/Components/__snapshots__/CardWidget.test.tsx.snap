// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardWidget /> snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa kHfmhG"
  >
    <h2
      class="sc-bwzfXH fNOhzS"
    >
      2020년 1분기 기대신작 애니
      <a
        class="sc-htpNat gcCwzB"
        href="/theme"
      >
        더보기
      </a>
    </h2>
    <div
      class="sc-bxivhb bUZGRr"
    >
      <div
        class="sc-ifAKCX kGuJUh"
      >
        <div
          class="sc-EHOje dfvRiZ"
          data-testid="card"
        >
          <div
            class="sc-gzVnrw cduVaT"
          >
            <img
              alt="no image"
              class="sc-htoDjs iphPbt"
              src="https://image.tmdb.org/t/p/w500//2CAL2433ZeIihfX1Hb2139CX0pW.jpg"
            />
          </div>
          <div
            class="sc-dnqmqq jPDSAr"
          >
            <span
              class="sc-iwsKbI hdcvZB"
              data-testid="genre"
            >
              35/18/10749
            </span>
            <a
              class="sc-gZMcBi cVhhuH"
              data-testid="title"
            >
              Dilwale Dulhania Le Jayenge
            </a>
          </div>
        </div>
        <div
          class="sc-EHOje duGfYM"
          data-testid="card"
        >
          <div
            class="sc-gzVnrw cduVaT"
          >
            <img
              alt="no image"
              class="sc-htoDjs iphPbt"
              src="https://image.tmdb.org/t/p/w500//9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg"
            />
          </div>
          <div
            class="sc-dnqmqq jPDSAr"
          >
            <span
              class="sc-iwsKbI hdcvZB"
              data-testid="genre"
            >
              80/18
            </span>
            <a
              class="sc-gZMcBi cVhhuH"
              data-testid="title"
            >
              The Shawshank Redemption
            </a>
          </div>
        </div>
        <div
          class="sc-EHOje duGfYM"
          data-testid="card"
        >
          <div
            class="sc-gzVnrw cduVaT"
          >
            <img
              alt="no image"
              class="sc-htoDjs iphPbt"
              src="https://image.tmdb.org/t/p/w500//rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg"
            />
          </div>
          <div
            class="sc-dnqmqq jPDSAr"
          >
            <span
              class="sc-iwsKbI hdcvZB"
              data-testid="genre"
            >
              18/80
            </span>
            <a
              class="sc-gZMcBi cVhhuH"
              data-testid="title"
            >
              The Godfather
            </a>
          </div>
        </div>
        <div
          class="sc-EHOje duGfYM"
          data-testid="card"
        >
          <div
            class="sc-gzVnrw cduVaT"
          >
            <img
              alt="no image"
              class="sc-htoDjs iphPbt"
              src="https://image.tmdb.org/t/p/w500//yPisjyLweCl1tbgwgtzBCNCBle.jpg"
            />
          </div>
          <div
            class="sc-dnqmqq jPDSAr"
          >
            <span
              class="sc-iwsKbI hdcvZB"
              data-testid="genre"
            >
              18/36/10752
            </span>
            <a
              class="sc-gZMcBi cVhhuH"
              data-testid="title"
            >
              Schindler's List
            </a>
          </div>
        </div>
        <div
          class="sc-EHOje duGfYM"
          data-testid="card"
        >
          <div
            class="sc-gzVnrw cduVaT"
          >
            <img
              alt="no image"
              class="sc-htoDjs iphPbt"
              src="https://image.tmdb.org/t/p/w500//7IiTTgloJzvGI1TAYymCfbfl3vT.jpg"
            />
          </div>
          <div
            class="sc-dnqmqq jPDSAr"
          >
            <span
              class="sc-iwsKbI hdcvZB"
              data-testid="genre"
            >
              35/18/53
            </span>
            <a
              class="sc-gZMcBi cVhhuH"
              data-testid="title"
            >
              Parasite
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
